egs/librispeech/ASR/zipformer/model.py

```diff
@@ -164,9 +164,9 @@ def forward_ctc(
 
         ctc_loss = torch.nn.functional.ctc_loss(
             log_probs=ctc_output.permute(1, 0, 2),  # (T, N, C)
-            targets=targets,
-            input_lengths=encoder_out_lens,
-            target_lengths=target_lengths,
+            targets=targets.to(torch.long),
```

Comment:
could you try
```
targets=targets.cpu(),
input_lengths=encoder_out_lens.cpu(),
target_lengths=target_lengths.cpu(),
```

It is a pity that PyTorch does not document clearly that only log_probs is required to be on GPU.